### DISCUSSION TITLE
Preprint on LiTI: using SLIM, tsdate, and msprime to infer population demography from lineage times.

### DISCUSSION THREAD

**Discussion by swamidass** [tskit-dev/tskit#1635] [Show and tell]
Jack Elliott-Higgins and I just released a preprint on some work we did using SLIM and tskit/tsdate, msprime, etc. We want to thank you all for making such an excellent code base that made this work possible. The manuscript should, hopefully, be under review soon. 

> ### Brief Population Bottlenecks Are Beyond The Genetic Streetlight
>
> Inferring human demographic history from extant genomes is an important goal of population genetics. To date, the sensitivity of coalescence-based methods in detecting population bottlenecks has not been well characterized. In this study, we find that brief bottlenecks, of just a few generations, are undetectable by current methods. A new approach to population inference, Lineage Time Inference (LiTI), uses data-derived windows to demarcate the limits of the genetic data. We find that a sharp population bottleneck at the time of the Youngest Toba Eruption, and also at more ancient timepoints in the human lineage, would be outside the genetic streetlight.

https://www.researchsquare.com/article/rs-817056/v1

There are two questions we had for you about this:

1. Would one of the main devs please verify that we are citing you in the preferred manner?

2. We have a an implementation of the LiTE algorithm, which infers population demography from a dated ARG.  Can we contribute this to the tskit code base? Where in the code base would it fit in?

Of course, any informal feedback on the preprint is much appreciated.

**Comment by jeromekelleher** [comment]
Hi @swamidass :wave: It's great to hear that the ts* tools have been useful to you, and the preprint looks interesting. Regarding the bottleneck model, I wonder if you considered the instantaneous bottleneck model in msprime described in [this tutorial](https://tskit.dev/tutorials/bottlenecks.html)?

To answer your questions:

1. The citations look good to me: thanks!
2. Tskit is for basic fundamental operations. We want to keep it reasonably compact and without too many dependencies, so we're hoping that people will build lots of tools *using* tskit, rather than trying to make it a universal library-of-everything. As such, inference methods such as ``tsinfer`` and ``tsdate`` are all in separate repositories and not part of the core library.

**Reply by swamidass** [reply]
Thanks @jeromekelleher.

As I understand it (and please correct me), backwards-times coalescence models like msprime use variants of the Kingman coalescent, so they all embed the assumption that the number of individuals in the population is much greater than the number of lineages.

When that assumption is violated (e.g. if there are 20 lineages active the generation after 4 individual bottleneck), then the approximations used to derive the Kingman coalescent break down. In those cases, doing a forward simulation with SLIM is the right approach, because a forward simulation doesn't have that same failure point.
 
In this paper, to make a point about detection limits, we are considering such low bottleneck sizes at such early timepoints that this sort of edge case is bound happen in several of the simulations we run.  That's why we didn't use msprime here.

Now, did I misunderstanding anything there about how misprime's instantaneous bottleneck model model works?

 ----

1. Glad the citation works.
2. Would it make sense to add to tsdate?
